core/collector.class.inc.php

```diff
@@ -717,32 +737,10 @@ public function Synchronize($iMaxChunkSize = 0)
 
 	public static function CallItopViaHttp($sUri, $aAdditionalData, $iTimeOut = -1)
 	{
-		$iCurrentTimeOut = ($iTimeOut !== -1) ? $iTimeOut : (int)Utils::GetConfigurationValue('itop_synchro_timeout', 600); // timeout in seconds, for a synchro to run
-
-		$sUrl = Utils::GetConfigurationValue('itop_url', '').$sUri;
-
-		$aResponseHeaders = null;
-
-		$aData = array_merge(
-			array(
-				'auth_user' => Utils::GetConfigurationValue('itop_login', ''),
-				'auth_pwd' => Utils::GetConfigurationValue('itop_password', ''),
-			),
-			$aAdditionalData
-		);
-
-		$aRawCurlOptions = Utils::GetConfigurationValue('curl_options', array(CURLOPT_SSLVERSION => CURL_SSLVERSION_SSLv3));
-		$aCurlOptions = array();
-		foreach ($aRawCurlOptions as $key => $value) {
-			// Convert strings like 'CURLOPT_SSLVERSION' to the value of the corresponding define i.e CURLOPT_SSLVERSION = 32 !
-			$iKey = (!is_numeric($key)) ? constant((string)$key) : (int)$key;
-			$iValue = (!is_numeric($value)) ? constant((string)$value) : (int)$value;
-			$aCurlOptions[$iKey] = $iValue;
+		if (null === static::$oCallItopService){
+			self::SetCallItopService(new CallItopService());
 		}
```

Comment:
No added value to call the method
```suggestion
		if (null === static::$oCallItopService){
			static::$oCallItopService = new CallItopService();
		}
```
Otherwise
```suggestion
		if (null === static::$oCallItopService){
			static::SetCallItopService(new CallItopService());
		}
```

Reply:
+1